test/integration/testExecute.js

```diff
@@ -213,14 +218,45 @@ describe('Execute test - variant', function ()
   var testCases =
     [
       {
-        name: 'raw xml',
+        name: 'xml_single_element',
+        type: 'XML',
+        fileExtension: '.xml',
+        sampleData: `<${TEST_HEADER}>${TEST_XML_VAL}</${TEST_HEADER}>`
+      },
+      {
+        name: 'xml_single_element_with_attribute',
+        type: 'XML',
+        fileExtension: '.xml',
+        sampleData: `<${TEST_HEADER} ${TEST_ATTRIBUTE_NAME}=${TEST_ATTRIBUTE_VALUE}>${TEST_XML_VAL}</${TEST_HEADER}>`,
+        ignoreAttributes: false
+      },
+      {
+        name: 'xml_with_attribute_and_custom_parser',
         type: 'XML',
         fileExtension: '.xml',
+        sampleData: `<node><${TEST_HEADER} ${TEST_ATTRIBUTE_NAME}=${TEST_ATTRIBUTE_VALUE}>${TEST_XML_VAL}</${TEST_HEADER}></node>`,
+        ignoreAttributes: false,
+        attributeNamePrefix: '##'
       },
       {
-        name: 'raw json',
+        name: 'xml_skip_attributes',
+        type: 'XML',
+        fileExtension: '.xml',
+        sampleData: `<node><${TEST_HEADER} ${TEST_ATTRIBUTE_NAME}=${TEST_ATTRIBUTE_VALUE}>${TEST_XML_VAL}</${TEST_HEADER}></node>`
+      },
+      {
+        name: 'xml_attribute_only',
+        type: 'XML',
+        fileExtension: '.xml',
+        sampleData: `<node><${TEST_HEADER} ${TEST_ATTRIBUTE_NAME}=${TEST_ATTRIBUTE_VALUE}></${TEST_HEADER}></node>`,
```

Comment:
let's add also test with `<node><${TEST_HEADER} ${TEST_ATTRIBUTE_NAME}=${TEST_ATTRIBUTE_VALUE}></${TEST_HEADER} />` since existing case could be parsed as empty text, not without body at all

Reply:
Added